app/models/user.rb

```diff
@@ -624,4 +625,13 @@ def can_send_confirmation_email
   rescue RateLimitChecker::LimitReached => e
     errors.add(:email, "confirmation could not be sent. #{e.message}")
   end
+
+  def update_rate_limit
+    return unless persisted?
```

Comment:
Nice! I would've easily overlooked this

Reply:
I missed it with the confirmation email which is why I caught it here. Experience really is directly related to the number of mistakes you make 😅